dataverse-ext-common/src/main/java/edu/harvard/iq/dataverse/importer/metadata/ResultField.java

```diff
@@ -0,0 +1,64 @@
+package edu.harvard.iq.dataverse.importer.metadata;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Container for name and value fields, but with the ability to contain also child elements, so
+ * it can be used to build tree-like structures.
+ */
+public class ResultField {
+    private String name;
+    private String value;
+    private List<ResultField> children;
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    private ResultField(String name, String value) {
+        this.name = name;
+        this.value = value;
+        this.children = Collections.emptyList();
+    }
+
+    private ResultField(String name, ResultField... children) {
+        this.name = name;
+        this.value = StringUtils.EMPTY;
+        this.children = children == null || children.length == 0
+                ? Collections.emptyList()
+                : Arrays.stream(children).collect(Collectors.toList());
+    }
+
+    // -------------------- GETTERS --------------------
+
+    public String getName() {
+        return name;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public List<ResultField> getChildren() {
+        return children;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    public static ResultField of(String name, String value) {
+        return new ResultField(name, value);
+    }
+
+    public static ResultField of(String name, ResultField... children) {
```

Comment:
I would also add constructor without children, atm if I wanna Result field without children I would need to do `ResultField.of(metadataFilter.getName(), (ResultField) null)` which is kinda unfriendly.

Ps. There are tests missing as mentioned in the main comment.
Ps2. Sorry for additional comments but there are some thoughts popping out while working alongside the code.

Reply:
But you can do it already – as the last parameter is vararg, you can write `ResultField.of("…")` and it will work.
Concerning tests: the problem is that the classes that should be tested will require rather complicated mocking and elaborate test cases, so I decided to postpone them to next steps, as the deadline is near.

Reply:
You are right, there is also another problem, was it intended to make children Immutable? I have to make a lot of operation in the parser in order to constantly copy lists around. In terms of tests and deadlines I will also ask Krzysiek. Personally I think tests are bounded with functionality and it is not something separate to be taken care of later (maybe if we are talking about some extensive Integration or Contract tests).

Reply:
1. I wanted to make the whole `ResultField` immutable, however I see, that I had to break immutability for one case. I think there are two ways to go:
- I can make this fully mutable, or
- I can convert it to immutable object, and write method(s) to merge it with another returning new immutable object.
2. I think that it's good idea to talk with Krzysiek, as I estimate that writing meaningful tests will consume 1, max 1.5 days, so if we can afford that amount of time now I see no other obstacles then.

Reply:
1. It can be immutable IMO, just with helper methods as you mentioned.